Signal/src/environment/OWSPreferences.m

```diff
@@ -0,0 +1,317 @@
+NSString *const OWSPreferencesKeyHasSentAMessage = @"User has sent a message";
+NSString *const OWSPreferencesKeyHasArchivedAMessage = @"User archived a message";
+NSString *const OWSPreferencesKeyPlaySoundInForeground = @"NotificationSoundInForeground";
+NSString *const OWSPreferencesKeyHasRegisteredVoipPush = @"VOIPPushEnabled";
+NSString *const OWSPreferencesKeyLastRecordedPushToken = @"LastRecordedPushToken";
+NSString *const OWSPreferencesKeyLastRecordedVoipToken = @"LastRecordedVoipToken";
+NSString *const OWSPreferencesKeyCallKitEnabled = @"CallKitEnabled";
+NSString *const OWSPreferencesKeyCallKitPrivacyEnabled = @"CallKitPrivacyEnabled";
+NSString *const OWSPreferencesKeyCallsHideIPAddress = @"CallsHideIPAddress";
+NSString *const OWSPreferencesKeyHasDeclinedNoContactsView = @"hasDeclinedNoContactsView";
+NSString *const OWSPreferencesKeyIOSUpgradeNagVersion = @"iOSUpgradeNagVersion";
+NSString *const OWSPreferencesKeyAreReadReceiptsEnabled = @"areReadReceiptsEnabled";
+
+@implementation OWSPreferences
+
+- (instancetype)init
+{
+    self = [super init];
+    if (!self) {
+        return self;
+    }
+
+    OWSSingletonAssert();
+
+    return self;
+}
+
+#pragma mark - Helpers
+
+- (void)clear {
+    @synchronized(self) {
+        NSString *appDomain = NSBundle.mainBundle.bundleIdentifier;
+        [NSUserDefaults.standardUserDefaults removePersistentDomainForName:appDomain];
+    }
+}
+
+- (nullable id)tryGetValueForKey:(NSString *)key
+{
+    ows_require(key != nil);
+    return [TSStorageManager.sharedManager objectForKey:key inCollection:OWSPreferencesSignalDatabaseCollection];
+}
+
+- (void)setValueForKey:(NSString *)key toValue:(nullable id)value
+{
+    ows_require(key != nil);
+
+    [TSStorageManager.sharedManager setObject:value forKey:key inCollection:OWSPreferencesSignalDatabaseCollection];
+}
+
+#pragma mark - Specific Preferences
+
+- (NSTimeInterval)getCachedOrDefaultDesiredBufferDepth
+{
+    id v = [self tryGetValueForKey:OWSPreferencesKeyCallStreamDESBufferLevel];
+    if (v == nil)
+        return OWSPreferencesDefaultCallStreamDESBufferLevel;
+    return [v doubleValue];
+}
+
+- (void)setCachedDesiredBufferDepth:(double)value
+{
+    ows_require(value >= 0);
+    [self setValueForKey:OWSPreferencesKeyCallStreamDESBufferLevel toValue:@(value)];
+}
+
+- (BOOL)screenSecurityIsEnabled
+{
+    NSNumber *preference = [self tryGetValueForKey:OWSPreferencesKeyScreenSecurity];
+    return preference ? [preference boolValue] : YES;
+}
+
+- (BOOL)getHasSentAMessage
+{
+    NSNumber *preference = [self tryGetValueForKey:OWSPreferencesKeyHasSentAMessage];
+    if (preference) {
+        return [preference boolValue];
+    } else {
+        return NO;
+    }
+}
+
+- (BOOL)getHasArchivedAMessage
+{
+    NSNumber *preference = [self tryGetValueForKey:OWSPreferencesKeyHasArchivedAMessage];
+    if (preference) {
+        return [preference boolValue];
+    } else {
+        return NO;
+    }
+}
+
+- (BOOL)hasRegisteredVOIPPush
```

Comment:
doesn't look like we're using this either.

Reply:
Done.